src/test/java/com/amit/electronic/store/controller/UserControllerTest.java

```diff
@@ -0,0 +1,127 @@
+package com.amit.electronic.store.controller;
+
+import com.amit.electronic.store.entity.User;
+import com.amit.electronic.store.service.UserServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.util.Collections;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class UserControllerTest {
+
+    @Mock
+    private UserServiceImpl userService;
+
+    @InjectMocks
+    private UserController userController;
+
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        mockMvc = MockMvcBuilders.standaloneSetup(userController).build();
+    }
+
+    @Test
+    void testHome() throws Exception {
+        mockMvc.perform(get("/store"))
+                .andExpect(status().isOk())
+                .andExpect(content().string("Welcome to Electronic Store, fellas!"));
+    }
+
+    @Test
+    void testCreateUser() throws Exception {
+        User user = new User(); // Create a user object for testing
+        when(userService.create(user)).thenReturn(user);
+
+        mockMvc.perform(post("/store/create")
```

Comment:
response is not asserted in any of these tests. How do we ensure that these are working correctly and would help in identifying regressions when there are changes to this in the future?

Reply:
mockMvc is performing the assertion when it calls perform method and expect the result and status